docs/design/proposals/creds-from-kms.md

```diff
@@ -0,0 +1,130 @@
+# Fetching Ceph user credentials from KMS
+
+## Proposal
+
+Ceph-CSI supports several KMS implementations for storing and
+fetching DEKs (Data Encryption Keys) used for RBD volume encryption.
+The focus of this proposal is to leverage existing KMS implementation
+to support obtaining Ceph user keyring used while
+creating, deleting, mapping and resizing RBD volumes.
+
+### Benefits
+
+- For scenarios where Ceph is deployed externally, either as
+  a standalone or through Rook, one needs to obtain the Ceph keyring(s)
+  and manually create the K8s secrets needed by Ceph-CSI. With this feature,
+  sensitive keyring(s) can be stored securely in an
+  external secret management system such as Hashicorp Vault,
+  AWS Secret Manager, etc., and have Ceph-CSI pull these
+  keys directly from them.
+- More secure compared to K8s secrets,
+  which are relatively easy to access and
+  store sensitive secrets as unencrypted based64 encoded text.
+
+### Drawbacks
+
+- Adds additional overhead as on each CSI RPC call
+  a request is made to KMS for fetching credentials.
+- Possible risk of hitting the KMS rate limit
+
+## Extending existing implementation
+
+- KMS ID is  provided as part of the provisioner,
+  node-stage, and controller-expand secrets, along with Ceph `userID`, as shown below.
+  Having this information as part of *StorageClass* parameters,
+  like using the `encryptionKMSID` key for volume encryption,
+  wouldn't work as these parameters are not passed
+  to all CSI RPCs, e.g. `DeleteVolume`.
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: csi-rbd-provisioner
+stringData:
+  userID: <ceph-user-id>
+  kmsID: <kms-id>
+ ```
+
+- A new ConfigMap with default name `ceph-csi-creds-kms-config` similar to
```

Comment:
why we need extra kms configmap cant we use existing kms configuration?

Reply:
Technically, we don't need a new config here. Given that structure of encryption and creds KMS config entries slight differ, specially due to the newly added `tenantNamespace` key and missing tenant sub config in case of creds, I thought it might make sense to keep them separate. 

I can change the section to use the same configuration if that makes sense. Please let me know what you think.

Reply:
Yes please

Reply:
I updated this section for reusing the existing KMS configuration.